core/drivers/cdns_uart.c

```diff
@@ -107,6 +108,7 @@ static const struct serial_ops cdns_uart_ops = {
 	.have_rx_data = cdns_uart_have_rx_data,
 	.putc = cdns_uart_putc,
 };
+KEEP_PAGER(cdns_uart_ops);
```

Comment:
Ah, sorry I forgot that we're printing all over the place also. The current proposal is correct, we'll need `.getchar` also from an secure interrupt handler.